morpheus/_lib/src/python_modules/messages.cpp

```diff
@@ -155,16 +157,40 @@ PYBIND11_MODULE(messages, m)
              py::return_value_policy::move)
         .def("get_meta_list", &MultiMessageInterfaceProxy::get_meta_list, py::return_value_policy::move);
 
-    py::class_<InferenceMemory, std::shared_ptr<InferenceMemory>>(m, "InferenceMemory")
-        .def_property_readonly("count", &InferenceMemoryInterfaceProxy::get_count);
+    py::class_<TensorMemory, std::shared_ptr<TensorMemory>>(m, "TensorMemory")
+        .def(py::init<>(&TensorMemoryInterfaceProxy::init),
+             py::arg("count"),
+             py::arg("tensors") = CupyUtil::py_tensor_map_t())
+        .def_readonly("count", &TensorMemory::count)
+        .def("get_tensors", &TensorMemoryInterfaceProxy::get_tensors, py::return_value_policy::move)
+        .def("set_tensors", &TensorMemoryInterfaceProxy::set_tensors, py::arg("tensors"))
+        .def("get_tensor", &TensorMemoryInterfaceProxy::get_tensor, py::arg("name"), py::return_value_policy::move)
+        .def("set_tensor", &TensorMemoryInterfaceProxy::set_tensor, py::arg("name"), py::arg("tensor"));
+
+    py::class_<InferenceMemory, TensorMemory, std::shared_ptr<InferenceMemory>>(m, "InferenceMemory")
+        .def(py::init<>(&InferenceMemoryInterfaceProxy::init),
+             py::arg("count"),
+             py::arg("tensors") = CupyUtil::py_tensor_map_t())
+        .def_readonly("count", &InferenceMemory::count)
+        .def("get_tensors", &InferenceMemoryInterfaceProxy::get_tensors, py::return_value_policy::move)
+        .def("set_tensors", &InferenceMemoryInterfaceProxy::set_tensors, py::arg("tensors"))
+        .def("get_tensor", &InferenceMemoryInterfaceProxy::get_tensor, py::arg("name"), py::return_value_policy::move)
+        .def("set_tensor", &InferenceMemoryInterfaceProxy::set_tensor, py::arg("name"), py::arg("tensor"));
```

Comment:
If you are deriving from `TensorMemory` there is no need to redefine these. It will use the base class.